src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonInputFormatter.cs

```diff
@@ -118,6 +120,19 @@ public class SystemTextJsonInputFormatter : TextInputFormatter, IInputFormatterE
             }
         }
 
+        private Exception WrapExceptionForModelState(JsonException jsonException)
+        {
+            if (!_jsonOptions.AllowSystemTextJsonInputFormatterExceptionMessages)
+            {
+                // This app is not opted-in to System.Text.Json messages, return the original exception.
```

Comment:
Nit:

```suggestion
                // This app is not opted-out of System.Text.Json messages, return the original exception.
```

Reply:
I took this from `Newtonsoft.Json`. If I'm not opted-in to `System.Text.Json` messages it means I shouldn't return `InputFormatterException` and the message will not be displayed to the client.